dm/dm/worker/server.go

```diff
@@ -441,26 +441,25 @@ func (s *Server) observeSourceBound(ctx context.Context, rev int64) error {
 }
 
 func (s *Server) doClose() {
-	s.cancel()
-	// close server in advance, stop receiving source bound and relay bound
```

Comment:
reason is here: // we should stop current sync first, other wise master may scheduler task on new worker while we are closing

i thinks if receiving source bound and relay bound when we are closing don't make things worse, if this worker is offline, those bound will be transfer to other free worker

Reply:
not all functions immediately response to `s.cancel()`, for example, some functions will use context of etcdClient which can cost 30s (so maybe we can close s.etcdClient in advance?). scheduler a new worker will add extra time ro wait doClose() return.

Also, please check if there are other cases that will cause doClose() slower after this change.

Reply:
> scheduler a new worker will add extra time ro wait doClose() return.

i thinks this is expected beheaviour, worker should cancel keepalive after sync task really stopped?

Reply:
I can understand the benefit of this change, originally it's fast and correct, now it's slow and correct.

Reply:
you are right, i  found i can call `s.cancel` with no wait https://github.com/pingcap/tiflow/pull/4035/commits/e881437b628ee2de07d95be1e6bd6d327b224d75